src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        decoder_inputs_embeds: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = False,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = False,
+        output_router_logits: Optional[bool] = False,
+    ) -> Union[Tuple[torch.FloatTensor], ModelOutput]:
+        r"""
+        Returns:
+            `ModelOutput` or `namedtuple` if `return_dict` returns ModelOutput insted of namedtuple
+
+        Example:
+
+        ```python
+        from transformers import AutoModel, AutoTokenizer
+
+        model = AutoModel.from_pretrained("Tanrei/GPTSAN-japanese")
+        tokenizer = AutoTokenizer.from_pretrained("Tanrei/GPTSAN-japanese")
+        x_tok = tokenizer.encode("武田信玄は、", return_tensors="pt")
+        model = model.cuda()
+        c = model.generate(x_tok.cuda(), max_new_tokens=50, random_seed=63)
+        tokenizer.decode(c[0])
+        "武田信玄は、戦国の頃より「智勇兼備」した英雄として織田信長に比されてきた戦国武将であり、..."
+        ```"""
+        NUM_TOKENS = self.config.vocab_size
+        SEP_TOKEN = NUM_TOKENS - 5
+        NOT_TOKEN = NUM_TOKENS - 4
+        device = self.position_embeddings.weight.device
+        if input_ids is None:
+            input_ids = torch.zeros([1, 1]).int().to(device)
+        num_pasts_contexts = 0
+        num_batch = input_ids.shape[0]
+        pasts = None
+        if past_key_values is not None:
+            num_pasts_contexts = past_key_values[0][0].shape[2]
+        elif self.config.d_spout and spout is not None:
+            num_pasts_contexts = 1
+
+        if num_precontext is not None:
+            if not (
+                len(num_precontext.shape) == 2 and num_precontext.shape[1] == 1
+            ):  # num_precontext Should be [batch,1]
+                raise ValueError("num_precontext should be [batch, 1] size.")
+            num_precontext = torch.reshape(num_precontext, [-1])
+        else:
+            num_precontext = torch.zeros([num_batch]).int().to(device)
+
+        num_input_contexts = input_ids.shape[1]
+        num_output_contexts = num_input_contexts + num_pasts_contexts
+
+        if past_key_values is not None:
+            if type(past_key_values) is tuple or type(past_key_values) is list:
+                op = []
+                for p in past_key_values:
+                    if type(p) is tuple or type(p) is list:
+                        p = torch.stack(p, dim=1)  # p Shuold be [batch, 2, heads, sequence, kernel]
+                    if p.shape != (
+                        num_batch,
+                        2,
+                        self.config.num_heads,
+                        num_pasts_contexts,
+                        self.config.d_model // self.config.num_heads,
+                    ):  # pasts Should be [batch, layer, 2, heads, sqquence, kernel]
+                        raise ValueError("past_key_values Tuple[Tuple[FloatTensor]]")
+                    op.append(p)
```

Comment:
No single letter variables. Also if you use the attention from transformers' codebase, this should not be needed.